### PR TITLE
force packaging versions

### LINKED ISSUE #473: [BUG] selinux RPM installation issues from copr
#### What happened:
SELINUX rpm is not working

#### What you expected to happen:
semodule --list | grep microshift to show microshift rules installed

#### How to reproduce it (as minimally and precisely as possible):
sudo dnf module -y enable cri-o:1.21
sudo dnf copr enable -y @redhat-et/microshift
sudo dnf install -y microshift firewalld
sudo systemctl enable crio --now
sudo systemctl enable microshift --now

#### Anything else we need to know?:
n/A

#### Environment:
- Microshift version (use `microshift version`): 4.8.0-0.okd-2021-10-10-030117
- Hardware configuration: amd 64
- OS (e.g: `cat /etc/os-release`): RHEL 8.4
- Kernel (e.g. `uname -a`):
- Others:

#### Relevant Logs 
```
  Preparing        :                                                                                                                                                                                                                      1/1 
  Reinstalling     : microshift-selinux-4.8.0-2021_11_19_125431.el8.noarch                                                                                                                                                                1/2 
  Running scriptlet: microshift-selinux-4.8.0-2021_11_19_125431.el8.noarch                                                                                                                                                                1/2 
libsemanage.semanage_pipe_data: Child process /usr/libexec/selinux/hll/pp failed with code: 255. (No such file or directory).
microshift: libsepol.policydb_read: policydb module version 20 does not match my version range 4-19
microshift: libsepol.sepol_module_package_read: invalid module in module package (at section 0)
microshift: Failed to read policy package
libsemanage.semanage_direct_commit: Failed to compile hll files into cil files.
 (No such file or directory).
/usr/sbin/semodule:  Failed!



```

### PR DESCRIPTION
**Pull request by cooktheryan**
Signed-off-by: Ryan Cook <rcook@redhat.com>

Force set versions.

Output RHEL 8 
```
Total                                                          79 MB/s |  36 MB     00:00     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                      1/1 
  Installing       : microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.noarch   1/2 
  Running scriptlet: microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.noarch   1/2 
  Installing       : microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.x86_64           2/2 
  Running scriptlet: microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.x86_64           2/2 
  Running scriptlet: microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.noarch   2/2 
  Running scriptlet: microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.x86_64           2/2 
  Verifying        : microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.x86_64           1/2 
  Verifying        : microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.noarch   2/2 
Installed products updated.

Installed:
  microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.x86_64                                  
  microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.el8.noarch                          

Complete!
```

Output Fedora 35 
```
Importing GPG key 0x26DF568C:
 Userid     : "cooktheryan_microshift (None) <cooktheryan#microshift@copr.fedorahosted.org>"
 Fingerprint: 6075 9B78 E731 4E58 7E4F E878 AA94 B3E7 26DF 568C
 From       : https://download.copr.fedorainfracloud.org/results/cooktheryan/microshift/pubkey.gpg
Is this ok [y/N]: y
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                             1/1 
  Installing       : microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.noarch                         1/2 
  Running scriptlet: microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.noarch                         1/2 
libsemanage.semanage_direct_install_info: A higher priority microshift module exists at priority 400 and will override the module currently being installed at priority 200.

  Installing       : microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.x86_64                                 2/2 
  Running scriptlet: microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.x86_64                                 2/2 
Removed /etc/systemd/system/multi-user.target.wants/microshift.service.

  Running scriptlet: microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.noarch                         2/2 
  Running scriptlet: microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.x86_64                                 2/2 
  Verifying        : microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.x86_64                                 1/2 
  Verifying        : microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.noarch                         2/2 

Installed:
  microshift-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.x86_64                                                        
  microshift-selinux-4.8.0-2022_01_04_175420_24_g8a3819ee.fc35.noarch                                                

Complete!
```
Closes #473

### CODE REVIEW

File: packaging/rpm/microshift.spec [hunk 1]
```diff
@@ -269,4 +273,4 @@ mv /usr/lib/systemd/system/microshift-containerized.service /usr/lib/systemd/sys
 - add missing BuildRequires on systemd and policycoreutils
```
Comment (mangelajo) [inline packaging/rpm/microshift.spec:274]: Let's add a packaging entry log here.

Please try it with make srpm it doesn't complain about the entrylog.

**Comment by mangelajo** [comment]
Awesome ryan!!! :-) thanks for looking on this.

**Comment by mangelajo** [comment]
image


works for me, but still there is some scriptlet error, the selinux warning goes away 

@cooktheryan is there a way to check if the module was installed?

**Comment by cooktheryan** [comment]
@mangelajo do ```semodule --list | grep microshift```

**Comment by cooktheryan** [comment]
@mangelajo did you have crio installed when you ran the above? I am not getting the scriptlet error on RHEL 8

**Comment by mangelajo** [comment]
> semodule --list | grep microshift
microshift

It's there.

I had crio installed. not sure what's going on but it's an improvement so let's merge IMHO.

I'll see if there's some way to debug the rpm install and figure that out.

**Pull request merged**